File: support/verifyunity/Assets/VerifyEnum.cs
using System;
using UnityEngine;

namespace verifycs {
    public static class VerifyEnum {
        public static void verifyEnum() {
            assert((int)SheeterEnum.VerifyEnum.Name0 == 0);
            assert((int)SheeterEnum.VerifyEnum.Name1 == 1);
            assert((int)SheeterEnum.VerifyEnum.Name2 == 2);
            Debug.Log("verify enum from: success");
        }

        private static void assert(bool condition) {
            if (condition == false)
                throw new Exception("verify enum: verify failed");
        }
    }
}

File: support/verifygo/verifyEnum.go
package main

import (
	"fmt"

	sheeterEnum "github.com/yinweli/Sheeter/support/example/enum/codeGo"
)

func verifyEnum() {
	assertEnum(sheeterEnum.ExampleEnum_Name0 == 0)
	assertEnum(sheeterEnum.ExampleEnum_Name1 == 1)
	assertEnum(sheeterEnum.ExampleEnum_Name2 == 2)
	fmt.Println("verify enum from: success")
}

func assertEnum(condition bool) {
	if condition == false {
		panic(fmt.Errorf("verify enum: verify failed"))
	} // if
}
